### PR TITLE
PyPI ready package

### PR DESCRIPTION
**Pull request by luca-drf**
- Fix tests and run them with pytest
- Update some functions and methods
- Automate test & deploy with Travis
- Fix setup.py to be PyPI compliant

### CODE REVIEW

File: src/test/test_yaml.py [hunk 1]
```diff
@@ -13,18 +13,20 @@
...
 
+here = pathlib.Path(__file__).parent.resolve()
 
 class YamlLoaderTest(unittest.TestCase):
     def test_parse_simple_str(self):
         home_dir = os.getenv("HOME")
         os.environ['EXPECTEDVAR'] = 'environmentVar'
         logging.info("Expected Home is %s" % home_dir)
-        config = load_config_yaml('./sample_config.yaml')
+        config = load_config_yaml(here/'sample_config.yaml')
```
Comment (marcoooo) [inline src/test/test_yaml.py:29]: Potential typo here?

Comment (luca-drf) [reply src/test/test_yaml.py:29]: Looks a bit too declarative for Python but that's actually how `pathlib` works. The `/` operator is overloaded for `Path` instances in order to work as concatenation: https://docs.python.org/3/library/pathlib.html#basic-use.

**Review by marcoooo — DISMISSED** [review]

**Review by marcoooo — COMMENTED** [review]
Since we already saw that 1.3 neede a 1.3.1, may be worth having a 1.4.0? Your opinion?

**Comment by luca-drf** [comment]
> Since we already saw that 1.3 neede a 1.3.1, may be worth having a 1.4.0? Your opinion?

I'm in favour, as long as we formalize how the version numbers are assigned. I'd adopt Semantic Versioning https://semver.org/

**Review by marcoooo — APPROVED** [review]

**Pull request merged**